sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusProcessorClientIntegrationTest.java

```diff
@@ -0,0 +1,204 @@
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled).setMessageId(messageId);
+        // The message should comeback after the client release the lock once because maxAutoLockRenewDuration is set
+        // by user.
+        CountDownLatch countDownLatch = new CountDownLatch(2);
+
+        // send the message
+        setSender(entityType, entityIndex, isSessionEnabled);
+        sendMessage(message).block(TIMEOUT);
+
+        if (isSessionEnabled) {
+            assertNotNull(sessionId, "'sessionId' should have been set.");
+            AmqpRetryOptions amqpRetryOptions = new AmqpRetryOptions().setTryTimeout(Duration.ofSeconds(lockTimeoutDurationSeconds));
+            processor = getSessionProcessorBuilder(false, entityType, entityIndex, false, amqpRetryOptions)
+                .maxAutoLockRenewDuration(expectedMaxAutoLockRenew)
+                .disableAutoComplete()
+                .processMessage(context -> processMessage(context, countDownLatch, messageId))
+                .processError(context -> processError(context, countDownLatch))
+                .buildProcessorClient();
+
+        } else {
+            this.processor = getProcessorBuilder(false, entityType, entityIndex, false)
+                .maxAutoLockRenewDuration(expectedMaxAutoLockRenew)
+                .disableAutoComplete()
+                .processMessage(context -> processMessage(context, countDownLatch, messageId))
+                .processError(context -> processError(context, countDownLatch))
+                .buildProcessorClient();
+        }
+
+        // Assert & Act
+        System.out.println("Starting the processor");
+        processor.start();
+
+        System.out.println("Listening for messages .. ");
+        if (countDownLatch.await(lockTimeoutDurationSeconds * 4, TimeUnit.SECONDS)) {
+            System.out.println("Message lock has been renewed. Now closing processor");
+        } else {
+            System.out.println("Message not arrived, closing processor.");
+            Assertions.fail("Message not arrived, closing processor.");
+        }
+
+        processor.close();
+    }
+
+    private void processMessage(ServiceBusReceivedMessageContext context, CountDownLatch countDownLatch, String expectedMessageId) {
+        ServiceBusReceivedMessage message = context.getMessage();
+        if (message.getMessageId().equals(expectedMessageId)) {
+            System.out.printf("Processing message. Session: %s, Sequence #: %s. Contents: %s%n", message.getMessageId(),
+                message.getSequenceNumber(), message.getBody());
+            countDownLatch.countDown();
+        } else {
+            System.out.printf("Received message, message id did not match. Session: %s, Sequence #: %s. Contents: %s%n", message.getMessageId(),
+                message.getSequenceNumber(), message.getBody());
+        }
```

Comment:
How does this ensure that the lock is renewed until `maxAutoLockRenewalDuration` and stops after? This will count down the latch immediately after the message is received resulting in the test passing without verifying the lock duration.

Reply:
Before the default message lock is set to 5 minute. And user was not able to overwrite it.
Now user set it to 15 seconds.

The message has to be received twice, first time we lock the message and we do not complete it, the message goes back to the queue and received again. The count down has to count twice for same message id. 
This ensure that we honor, user's maxAutoLockRenewalDuration instead of 5 minutes.

The fact that lock is released and message come back, that shows that the lock renew operation stops .